src/widget_type.h

```diff
@@ -364,13 +355,23 @@ enum NWidgetDisplay {
 };
 DECLARE_ENUM_AS_BIT_SET(NWidgetDisplay)
 
+/** Container with the data associated to a single widget. */
+struct WidgetData {
+	StringID string{};
+	SpriteID sprite{};
+	ArrowWidgetValues arrow_widget_type{};
+	ResizeWidgetValues resize_widget_type{};
+	uint8_t matrix_columns{};
+	uint8_t matrix_rows{};
+};
```

Comment:
Is being a std::variant in works? (I know that needs at least some StrongType stuff)

Reply:
I have not started on that yet. If we strong-type `StringID`, it would be relatively easy I guess.